File: src/main/java/com/hzj/myblog/service/BlogTypeService.java
package com.hzj.myblog.service;

import com.hzj.myblog.model.BlogType;

import java.util.List;

/**
 * 博客分类的service层接口
 *
 * @author hzj
 */
public interface BlogTypeService {
    /**
     * 添加博客分类
     *
     * @param blogType 分类信息
     */
    void addBlogType(BlogType blogType);

    /**
     * 删除博客分类
     *
     * @param id 博客分类
     */
    void deleteBlogType(Integer id);

    /**
     * 查找所有的博客分类
     *
     * @return 返回查找到的列表
     */
    List<BlogType> findAllBlogType();

    /**
     * 更新博客分类的信息
     *
     * @param blogType 博客分类的信息
     */
    void updateBlogType(BlogType blogType);
}

File: my-blog-web/src/app/@core/interface/blog-type.service.ts
import {Observable} from "rxjs";
import {ReturnModel} from "../../entity/return-model";
import {BlogType} from "../../model/blog-type";

export abstract class BlogTypeService {

  /**
   * 新增博客分类
   * @param typeName 博客分类
   */
  abstract addBlogType(typeName: string): Observable<ReturnModel<any>>;

  /**
   * 删除博客分类
   * @param blogTypeId 博客分类的id
   */
  abstract deleteBlogType(blogTypeId: number): Observable<ReturnModel<any>>;

  /**
   * 查找所有的博客分类
   */
  abstract findAllBlogType(): Observable<ReturnModel<BlogType[]>>;

  /**
   * 更新博客分类的信息
   * @param blogType 博客分类
   */
  abstract updateBlogType(blogType: BlogType): Observable<ReturnModel<BlogType>>;
}
